warehouse/query-core/src/main/java/datawave/query/planner/FederatedQueryPlanner.java

```diff
@@ -0,0 +1,172 @@
+package datawave.query.planner;
+
+import datawave.query.CloseableIterable;
+import datawave.query.config.FieldIndexHole;
+import datawave.query.config.ShardQueryConfiguration;
+import datawave.query.config.ValueIndexHole;
+import datawave.query.exceptions.DatawaveQueryException;
+import datawave.query.planner.pushdown.rules.PushDownRule;
+import datawave.query.tables.ScannerFactory;
+import datawave.query.util.MetadataHelper;
+import datawave.query.util.YearMonthDay;
+import datawave.util.time.DateHelper;
+import datawave.webservice.common.logging.ThreadConfigurableLogger;
+import datawave.webservice.query.Query;
+import datawave.webservice.query.configuration.GenericQueryConfiguration;
+import datawave.webservice.query.configuration.QueryData;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.iterators.SortedKeyValueIterator;
+import org.apache.accumulo.core.metadata.schema.MetadataSchema;
+import org.apache.commons.jexl2.parser.ASTJexlScript;
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+
+import java.util.*;
+
+public class FederatedQueryPlanner extends DefaultQueryPlanner {
+    
+    private static final Logger log = ThreadConfigurableLogger.getLogger(FederatedQueryPlanner.class);
+    {
+        log.setLevel(Level.DEBUG);
+    }
+    
+    public FederatedQueryPlanner() {
+        super();
+    }
+    
+    @Override
+    public FederatedQueryDataIterable process(GenericQueryConfiguration config, String query, Query settings, ScannerFactory scannerFactory)
+                    throws DatawaveQueryException {
+        
+        FederatedQueryDataIterable returnQueryData = new FederatedQueryDataIterable();
+        Date originalEndDate = config.getEndDate();
+        Date originalStartDate = config.getBeginDate();
+        TreeSet<YearMonthDay> holeDates;
+        
+        // The DefaultQueryPlanner.process needs to be called so that the FieldIndexHoles can be calculated
+        // and the queryData returned will be used if there are no index holes of any kind.
+        CloseableIterable<QueryData> queryData = super.process(config, query, settings, scannerFactory);
```

Comment:
The index holes should be pulled separately in this class and processed independently from the DefaultQueryPlanner.  The DefaultQueryPlanner process method can take many minutes to run when expanding fields against the global index.